Framework/Algorithms/src/XrayAbsorptionCorrection.cpp

```diff
@@ -0,0 +1,234 @@
+
+using namespace Mantid::API;
+using namespace Mantid::Geometry;
+using namespace Mantid::Kernel;
+using Mantid::DataObjects::Workspace2D;
+namespace PhysicalConstants = Mantid::PhysicalConstants;
+
+/// @cond
+namespace {
+
+constexpr double DEFAULT_ANGLE = 45.0;
+constexpr double DEFAULT_DETECTOR_DISTANCE = 10.0;
+
+} // namespace
+/// @endcond
+
+namespace Mantid {
+namespace Algorithms {
+
+DECLARE_ALGORITHM(XrayAbsorptionCorrection)
+
+/**
+ * Initialize the algorithm
+ */
+void XrayAbsorptionCorrection::init() {
+
+  auto wsValidator = std::make_shared<CompositeValidator>();
+
+  declareProperty(std::make_unique<WorkspaceProperty<>>(
+                      "InputWorkspace", "", Direction::Input, wsValidator),
+                  "The name of the input workspace.");
+
+  declareProperty(
+      std::make_unique<WorkspaceProperty<>>("MuonImplantationProfile", "",
+                                            Direction::Input, wsValidator),
+      "The name of the Muon Implantation Profile.");
+
+  declareProperty(std::make_unique<WorkspaceProperty<>>("OutputWorkspace", "",
+                                                        Direction::Output),
+                  "The name to use for the output workspace.");
+
+  auto positiveDouble = std::make_shared<Kernel::BoundedValidator<double>>();
+  positiveDouble->setLower(0);
+  declareProperty("DetectorAngle", DEFAULT_ANGLE, positiveDouble,
+                  "Angle between beam and Detector."
+                  "Range of normal values for detectors are :  "
+                  "Ge1 : 90-180 , Ge2 : 270-360 , Ge3 : 0 - 90 , Ge4 "
+                  ": 180 -270.",
+                  Direction::Input);
+
+  declareProperty("DetectorDistance", DEFAULT_DETECTOR_DISTANCE, positiveDouble,
+                  "Distance between detector and sample.", Direction::Input);
+}
+
+std::map<std::string, std::string> XrayAbsorptionCorrection::validateInputs() {
+  MatrixWorkspace_sptr inputWS = getProperty("InputWorkspace");
+  MatrixWorkspace_sptr muonProfile = getProperty("MuonImplantationProfile");
+  std::map<std::string, std::string> issues;
+  if (!inputWS->sample().getShape().hasValidShape()) {
+    issues["InputWorkspace"] = "Input workspace does not have a Sample";
+  } else {
+    auto material = inputWS->sample().getShape().material();
+    if (!material.hasValidxRayAttenuationOverride()) {
+      issues["InputWorkspace"] =
+          "Input workspace does not have a Xray Attenuation profile";
+    } else {
+      if (muonProfile->getNumberHistograms() != 1) {
+        issues["MuonImplantationProfile"] =
+            "Muon Implantation profile must have only one spectrum";
+      }
+    }
+  }
+  return issues;
+}
+
+/**
+ * Converts angles in degrees to angles in radians.
+ * @param degrees double of angle in degrees
+ * @return A double of angle in radians
+ */
+double XrayAbsorptionCorrection::degreesToRadians(double degrees) {
+  double radians = M_PI * (degrees / 180);
+  return radians;
```

Comment:
we can do this on one line. You should also explicitly use doubles.
```suggestion
  return  M_PI * (degrees / 180.0);
```
```suggestion
  return radians;
```